src/Grid.php

```diff
@@ -513,6 +513,32 @@ public function addModalAction($button, $title, \Closure $callback, $args = [])
         return $this->getActionButtons()->addModal($button, $title, $callback, $this, $args);
     }
 
+    /**
+     * Similar to addModalAction but apply to a multiple recors selection and display in menu.
+     * When menu item is clicked, modal is displayed with the $title and $callback is executed through
+     * VirtualPage.
+     *
+     * @param string|array|MenuItem            $item
+     * @param \Closure(View, array|null): void $callback
+     * @param array                            $args     extra url argument for callback
+     *
+     * @return Jquery
+     */
+    public function addModalBulkAction($item, \Closure $callback, $args = [])
+    {
+        if (is_string($item)) {
+            $item = ['title' => $item];
+        }
+
+        $modal = Modal::addTo($this->getOwner());
+
+        $modal->set(function (View $t) use ($callback) {
+            $callback($t, $t->stickyGet($this->name) ? explode(',', $t->stickyGet($this->name)) : null);
+        });
+
+        return $this->menu->addItem($item)->on('click', $modal->jsShow(array_merge([$this->name => $this->selection->jsChecked()], $args)), []);
```

Comment:
any reason why this return JsExpressionable and non-bulk method View?

Reply:
Should return View.

Reply:
Done